pkg/gadgets/snapshot/socket/tracer/bpf/socket-common.h

```diff
@@ -66,13 +73,36 @@ static __always_inline void socket_bpf_seq_print(struct seq_file *seq,
      * in this format:
      *
      * protocol: "TCP" or "UDP"
-     * IP addresses and ports: Hexadecimal in host-byte order.
+     * family: AF_INET
+     * IP addresses: Hexadecimal in network-byte order.
+     * ports: Hexadecimal in host-byte order.
      * state: Hexadecimal of https://github.com/torvalds/linux/blob/v5.13/include/net/tcp_states.h#L12-L24
      * ino: unsigned long.
      */
-    BPF_SEQ_PRINTF(seq, "%s %08X %04X %08X %04X %02X %lu\n",
-        protocol, bpf_ntohl(src), bpf_ntohs(srcp),
-        bpf_ntohl(dest), bpf_ntohs(destp), state, ino);
+    BPF_SEQ_PRINTF(seq, "%s %04X %08X %04X %08X %04X %02X %lu\n",
+        protocol, AF_INET, src, bpf_ntohs(srcp),
+        dest, bpf_ntohs(destp), state, ino);
+}
+
+static __always_inline void socket_bpf_seq_print_v6(struct seq_file *seq,
+                const char* protocol, const __u8 *src,
+                const __u16 srcp, const __u8 *dest,
+                const __u16 destp, const unsigned char state, long ino)
+{
+    /*
+     * Notice that client side program is expecting socket information exactly
+     * in this format:
+     *
+     * protocol: "TCP" or "UDP"
+     * family: AF_INET
+     * IP addresses: Printed as IPv6 notation (https://www.kernel.org/doc/html/latest/core-api/printk-formats.html#ipv6-addresses).
+     * ports: Hexadecimal in host-byte order.
+     * state: Hexadecimal of https://github.com/torvalds/linux/blob/v5.13/include/net/tcp_states.h#L12-L24
+     * ino: unsigned long.
+     */
+    BPF_SEQ_PRINTF(seq, "%s %04X %pI6 %04X %pI6 %04X %02X %lu\n",
+        protocol, AF_INET6, src, bpf_ntohs(srcp),
+        dest, bpf_ntohs(destp), state, ino);
```

Comment:
This is not fully related to your PR but at this point I'm wondering if we should switch to a binary representation (use bpf_seq_write as in pkg/gadgets/top/ebpf/piditer/bpf/pid_iter.bpf.c) instead of parsing a string. IMO that's more efficient and easier to handle. 

IIUC the advantage of using `BPF_SEQ_PRINTF` is that one can pin the iterator and then any application can ready it getting a text representation, but for us this is not a valuable use case as it's the gadget itself reading and parsing it.

Reply:
In general I would agree when the struct has a fixed size, which is the case in all our current snapshot gadgets, but not in the new "snapshot file" gadget using `bpf_seq_printf_btf()` to output lengthy text.

Other considerations:
- Ideally, if we change the parsing, it should be something practical for containerized gadgets (using either BTF or some generic text parsing method)
- We still have to fix the issue with the `comm` field contains a `\n`.